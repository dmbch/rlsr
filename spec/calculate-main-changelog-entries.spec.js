/* eslint-env node, jest */
const R = require('ramda');

const calculateMainChangelogEntries = require('../src/transform/calculate-main-changelog-entries');

const getEnv = require('./fixtures/env-small.fixture');
const getMessage = require('./fixtures/message-small.fixture');

describe('calculateMainChangelogEntries()', () => {
  const envProto = getEnv(
    {
      one: {
        name: 'one',
        version: '1.2.3',
        rlsr: {
          hasBump: true,
          determinedIncrementLevel: 0,
          messages: [getMessage('foo')],
          relatedMessages: [],
        },
      },
    },
    []
  );
  envProto.mainPackage = {
    version: '2.3.4',
  };

  it('adds a fresh changelog if nothing was available before', () => {
    Date.now = jest.fn(() => 0);
    const env = R.clone(envProto);
    const exp = calculateMainChangelogEntries(env);
    const changelog = exp.changelog;
    R.values(changelog);

    expect(changelog).toMatchSnapshot();
  });

  it('adds an entry to an existing changelog', () => {
    Date.now = jest.fn(() => 0);
    const env = R.clone(envProto);
    env.changelog = { '1.2.3': 'foo' };
    const exp = calculateMainChangelogEntries(env);
    const changelog = exp.changelog;
    R.values(changelog);

    expect(changelog).toMatchSnapshot();

    // Check if the order is correct
    expect(Object.keys(changelog)).toEqual(['2.3.4', '1.2.3']);
  });
});
