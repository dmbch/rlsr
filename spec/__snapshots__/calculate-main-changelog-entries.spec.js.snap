// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateMainChangelogEntries() adds a fresh changelog if nothing was available before 1`] = `
Object {
  "2.3.4": Object {
    "changes": Object {
      "one": Object {
        "determinedIncrementLevel": "patch",
        "messages": Array [
          Object {
            "affected": Array [
              "one",
            ],
            "body": undefined,
            "subject": "foo",
            "type": "fix",
          },
        ],
        "version": "1.2.3",
      },
    },
    "date": "Thu Jan 01 1970 01:00:00 GMT+0100 (CET)",
  },
}
`;

exports[`calculateMainChangelogEntries() adds an entry to an existing changelog 1`] = `
Object {
  "1.2.3": "foo",
  "2.3.4": Object {
    "changes": Object {
      "one": Object {
        "determinedIncrementLevel": "patch",
        "messages": Array [
          Object {
            "affected": Array [
              "one",
            ],
            "body": undefined,
            "subject": "foo",
            "type": "fix",
          },
        ],
        "version": "1.2.3",
      },
    },
    "date": "Thu Jan 01 1970 01:00:00 GMT+0100 (CET)",
  },
}
`;
